### PR TITLE
Change CAN name rules

### PR DESCRIPTION
**Pull request by quocle108**
## Change Description

* A account length can be between 6 and 12 (both inclusive).
* Available characters are a to z and 1 to 5.
* Users whose account length is between 6 and 12 can't create other accounts ending with dot and their username.
* The premium name market is closed at the time of launching the mainnet.
* The premium name market to be opened later will allow to bid username less than 6 and to use it to create other accounts ending with dot and the username. 

## Deployment Changes
- [ ] Deployment Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by quocle108** [comment]
@danielAlvess More info about eos name:

Account Name Design
* 12 characters max
* May contain: a-z, 1-5, or .
* May not end with .
Names are encoded in base-32:
* . = 0
* 1-5 = 1 - 5
* a-z = 6 - 31
The 5 MSBs of the uint64 contain the first character. Unused characters are replaced with . (0).

**Comment by quocle108** [comment]
@danielAlvess FYI:

**Account Name Design**

* 12 characters max
* May contain: a-z, 1-5, or .
* May not end with .

**Names are encoded in base-32:**
* . = 0
* 1-5 = 1 - 5
* a-z = 6 - 31
The 5 MSBs of the uint64 contain the first character. Unused characters are replaced with . (0).

**Converting Strings**

An account name can be converted to or from strings as follows:
"dan"          <=> "dan........."
"dan.larimer"  <=> "dan.larimer."

Refer:

https://github.com/EOSIO/eos/issues/56
https://developers.eos.io/welcome/latest/protocol/accounts_and_permissions

**Comment by danielvo11** [comment]
I'm reviewing!.

File: contracts/eosio.system/src/eosio.system.cpp [hunk 1]
```diff
@@ -332,18 +332,23 @@ namespace eosiosystem {
            has_dot |= !(tmp & 0x1f);
            tmp >>= 5;
          }
-         if( has_dot ) { // or is less than 12 characters
+
+         bool has_below_6_chars = (newact.value & 0x7FFFFFFFF0ull) == 0;
```
Comment (danielvo11) [inline contracts/eosio.system/src/eosio.system.cpp:336]: 0xFFFFFFFFFF0

**Review by danielvo11 — APPROVED** [review]

**Pull request merged**